crates/database/src/states/state.rs

```diff
@@ -314,6 +315,102 @@ impl<DB: Database> DatabaseCommit for State<DB> {
     }
 }
 
+impl<DB: DatabaseRef> DatabaseRef for State<DB> {
+    type Error = DB::Error;
+
+    fn basic_ref(&self, address: Address) -> Result<Option<AccountInfo>, Self::Error> {
+        match self.cache.accounts.get(&address) {
+            Some(account) => {
+                // Account is already in cache
+                return Ok(account.account_info());
+            }
+            None => {
+                if self.use_preloaded_bundle {
+                    // Check if account is in bundle state
+                    if let Some(account) = self.bundle_state.account(&address) {
+                        return Ok(CacheAccount::from(account.clone()).account_info());
```

Comment:
Try not to clone full account, but to just extract account info from it. Cloning is expensive as it probably contain storage.